query/src/common/hashtable/hash_table.rs

```diff
@@ -35,6 +35,16 @@ pub struct HashTable<Key: HashTableKeyable, Entity: HashTableEntity<Key>, Grower
     generics_hold: PhantomData<Key>,
 }
 
+unsafe impl<Key: HashTableKeyable, Entity: HashTableEntity<Key>, Grower: HashTableGrower> Send
+    for HashTable<Key, Entity, Grower>
+{
+}
+
+unsafe impl<Key: HashTableKeyable, Entity: HashTableEntity<Key>, Grower: HashTableGrower> Sync
+    for HashTable<Key, Entity, Grower>
+{
+}
```

Comment:
Maybe we should remove this?

Reply:
It seems not, because `HashJoinState` implements send & sync

Reply:
`HashJoinState` is required to be `Sync + Send`, which is satisfied by `Arc<Mutex<HashTable>>`. It's not safe to impl `Send` and `Sync` for `HashTable` since it can not be shared between threads without a mutex guard.